plugins/module_utils/network/sonic/argspec/snmp/snmp.py

```diff
@@ -0,0 +1,227 @@
+                            },
+                            'type': 'list'
+                        },
+                        'name': {
+                            'required': True,
+                            'type': 'str'
+                        }
+                    },
+                    'type': 'list'
+                },
+                'host': {
+                    'elements': 'dict',
+                    'options': {
+                        'community': {'type': 'str'},
+                        'ip': {
+                            'required': True,
+                            'type': 'str'
+                        },
+                        'port': {'type': 'int'},
+                        'retries': {'type': 'int'},
+                        'source_interface': {'type': 'str'},
+                        'tag': {
+                            'choices': ['inform', 'trap'],
+                            'type': 'str'
+                        },
+                        'timeout': {'type': 'int'},
+                        'user': {
+                            'options': {
+                                'name': {
+                                    'required': True,
+                                    'type': 'str'
+                                },
+                                'security_level': {
+                                    'choices': [
+                                        'auth',
+                                        'noauth',
+                                        'priv'
+                                    ],
+                                    'type': 'str'
+                                }
+                            },
+                            'type': 'dict'
+                        },
+                        'vrf': {'type': 'str'}
+                    },
+                    'type': 'list'
+                },
+                'location': {'type': 'str'},
+                'user': {
+                    'elements': 'dict',
+                    'options': {
+                        'group': {'type': 'str'},
+                        'name': {
+                            'required': True,
+                            'type': 'str'
+                        },
+                        'auth': {
+                            'options': {
+                                'auth_type': {
+                                    'choices': [
+                                        'md5',
+                                        'sha',
+                                        'sha2-256',
+                                        'sha2-384',
+                                        'sha2-512'
+                                    ],
+                                    'type': 'str'
+                                },
+                                'key': {
+                                    'type': 'str',
+                                    'no_log': True
```

Comment:
Support of idempotency for encrypted keys would be possible if the no_log value is changed to False.

```suggestion
                                    'no_log': False
```